### PR TITLE
feat(developer) marker steps 🙀 

### PR DESCRIPTION
**Pull request by srl295**
- validate markers
- err if markers are 'matched' but not 'emitted'
- assumes string variables are 'emitted'
- binary format: sets the 'vars.markers' list (and basic.txt to match)

#9119 

@keymanapp-test-bot skip

### CODE REVIEW

File: core/src/ldml/C9134_ldml_markers.md [hunk 1]
```diff
@@ -27,7 +27,6 @@ Markers can appear in both 'emitting' and 'matching-only' areas:
 #### Match only
 
 - `transform from=` to match markers
-- `transform after=` to match markers
```
Comment (srl295) [inline core/src/ldml/C9134_ldml_markers.md:30]: where did _this_ come from?

File: developer/src/kmc-ldml/src/compiler/vars.ts [hunk 2]
```diff
@@ -130,9 +133,60 @@ export class VarsCompiler extends SectionCompiler {
...
   }
 
+  private collectMarkers(emitMarkers : Set<string>, matchMarkers : Set<string>) : boolean {
+    let valid = true;
+
+    // call our friends to validate
+    valid = this.validateVarsMarkers(this.keyboard, emitMarkers, matchMarkers) && valid; // accumulate validity
+    valid = KeysCompiler.validateMarkers(this.keyboard, emitMarkers, matchMarkers) && valid; // accumulate validity
+    valid = TransformCompiler.validateMarkers(this.keyboard, emitMarkers, matchMarkers) && valid; // accumulate validity
+    valid = DispCompiler.validateMarkers(this.keyboard, emitMarkers, matchMarkers) && valid; // accumulate validity
```
Comment (srl295) [inline developer/src/kmc-ldml/src/compiler/vars.ts:149]: so i'm trying to keep the locality of code here.  so `disp.ts` is still responsible for understanding the `<display>` element etc

**Review by mcdurdin — APPROVED** [review]
LGTM

File: developer/src/kmc-ldml/src/compiler/vars.ts [hunk 3]
```diff
@@ -130,9 +134,56 @@ export class VarsCompiler extends SectionCompiler {
...
+      valid = false;
+    }
     return valid;
   }
 
+  validateVarsMarkers(keyboard: LDMLKeyboard.LKKeyboard, mt : MarkerTracker) : boolean {
+    keyboard?.variables?.string?.forEach(({value}) =>
+          mt.add(MarkerUse.variable, MarkerParser.allReferences(value)));
+    return true;
+  }
```
[the remark this answers is not in the document]
Comment (srl295) [reply developer/src/kmc-ldml/src/compiler/vars.ts:185]: It supports validation by checking existence. Though I thought maybe there should be a second looser regex that catches invalid marker ids ?

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 17.0.151-alpha](https://keyman.com/downloads/releases/alpha/17.0.151)

**Pull request merged**